### PR TITLE
Add remove mothod to the component state

### PR DESCRIPTION
**Pull request by vdubr**

### CODE REVIEW

File: src/state/Components/reducers.js [hunk 1]
```diff
@@ -25,47 +27,56 @@ function update(state, componentKey, update) {
...
  */
-function set(state, component, path, value) {
-	if (component && path) {
-		const pathParams = path.split('.');
-		return {
-			...state,
-			[component]: setHelper(state[component], pathParams, value),
-		};
+function remove(state, componentKey, path = '') {
+	const pathForComponent = !path || path === null || path === '';
```
Comment (vlach1989) [inline src/state/Components/reducers.js:38]: Add comments to describe this

File: src/state/Components/helpers.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Set value on given path. If value is undefined, then delete property on path.
+ * @param state {Object}
+ * @param path {Array}
+ * @param value {*}
+ * @return {Object}
+ */
+function setHelper(state, path, value) {
+	let remainingPath = [...path];
```
Comment (vlach1989) [inline src/state/Components/helpers.js:9]: What if path is undefined?

**Review by vlach1989 — CHANGES_REQUESTED** [review]

File: src/state/Components/reducers.js (hunk 1 above)
Comment (vdubr) [reply src/state/Components/reducers.js:38]: Comments are in method docs.

File: src/state/Components/helpers.js (hunk 2 above)
Comment (vdubr) [reply src/state/Components/helpers.js]: fixed: f5621ef73fa5e2dd617500cf4fe70c44803d1042

**Review by vlach1989 — APPROVED** [review]

**Pull request merged**